### PR TITLE
simplify dispatcher if-elif

### PR DESCRIPTION
**Pull request by pmeier**
Fixes https://github.com/pytorch/vision/pull/7082#discussion_r1069077513.

cc @vfdev-5 @datumbox @bjuncek

### CODE REVIEW

**Comment by pmeier** [comment]
This works at runtime and with JIT, but `mypy` is not happy. Since we completely ignore PIL

https://github.com/pytorch/vision/blob/035d99f1e7f7206a8708e76f4568aac0c3b1346d/mypy.ini#L120-L122

it treats all annotations with `PIL.Image.Image` as `Any`. Meaning, although we correctly annotate

https://github.com/pytorch/vision/blob/035d99f1e7f7206a8708e76f4568aac0c3b1346d/torchvision/prototype/transforms/functional/_misc.py#L158-L160

for `mypy` this looks like `def gaussian_blur_image_pil(...) -> Any`. Subsequently, it complains that we return `Any` from a function annotated with `torch.Tensor`:

https://github.com/pytorch/vision/blob/035d99f1e7f7206a8708e76f4568aac0c3b1346d/torchvision/prototype/transforms/functional/_misc.py#L184-L185

I'm not sure how this worked before, since in any case due to our JIT annotation, neither `PIL.Image.Image` nor `Any` should have been allowed. Maybe the convoluted check was designed in a way to "confuse" `mypy` to not complain?

I see three ways forward here:

1. Keep everything as is, since it works. Given the situation that might be brittle, since `mypy` maybe at some point can look through the convoluted check and will complain then.
2. Add a `# type: ignore` on all (27) of the PIL returns in the dispatchers.
3. Drop the strict
   https://github.com/pytorch/vision/blob/035d99f1e7f7206a8708e76f4568aac0c3b1346d/mypy.ini#L35
   configuration.

My preference is 3. -> 2. -> 1.

**Comment by NicolasHug** [comment]
Thanks for looking into it Philip. I agree with the order of preference and I'm happy with 3.

File: torchvision/prototype/transforms/_type_conversion.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ class ToImageTensor(Transform):
     def _transform(
         self, inpt: Union[torch.Tensor, PIL.Image.Image, np.ndarray], params: Dict[str, Any]
     ) -> datapoints.Image:
-        return F.to_image_tensor(inpt)  # type: ignore[no-any-return]
```
Comment (pmeier) [inline torchvision/prototype/transforms/_type_conversion.py:49]: Due to the relaxed strictness, this one (and the ones below) became obsolete.

**Review by NicolasHug — APPROVED** [review]
Thanks a lot Philip, LGTM.

I have 2 follow-up questions / comments, which we can address (or not) in other PRs:

- As discussed offline, JIT "isn't supported" for the datapoints on these dispatchers. What we mean by "not supported" is that the datapoints will be treated as a pure Tensors, and the output will be a Tensor too. I assume this can lead to unexpected results and confusion, e.g.:

    ```py
    mask = Mask(...)
    out = T(mask)
    # - unaware user expects a Mask but gets a Tensor
    # - `T` treats mask as an image, not as a mask!
    #    Might result in different behaviour (very bad) or confusing error message?
    ```

    so: **should we fail loudly** with a good error message in the disaptchers when `is_scripting()` is True and `is_simple_tensor()` is False?


- Side note: should we be more lenient with `mypy` checks in the prototype area?
We introduced strict checks in https://github.com/pytorch/vision/pull/4513 (and I approved the PR, so I'll concede I'm sort of rebutting myself here).
But the the prototype area is where we want to be able to move fast, and try potentially crazy things without having to worry too much about type checks / docs / super clean tests. In this specific case, the introduction of `warn_return_any` could have been the reason for these convoluted checks in the first place which complicates the code. Perhaps we may want to avoid to relax the mypy requirements in order to experiment faster in the prototype area?

**Comment by pmeier** [comment]
1. I'm looking into it. `is_simple_tensor` seems to not behave as we want it to be when scripting. Will keep you posted.
2. Yes, I'm ok with that.

**Comment by pmeier** [comment]
`mypy` is happy here: https://app.circleci.com/jobs/github/pytorch/vision/1807247

**Pull request merged**